homeassistant/helpers/service.py

```diff
@@ -355,15 +355,22 @@ async def async_extract_referenced_entity_ids(
     # Find devices for this area
     selected.referenced_devices.update(selector.device_ids)
     for device_entry in dev_reg.devices.values():
-        if device_entry.area_id in selector.area_ids:
+        if (
+            device_entry.area_id in selector.area_ids
+            or device_entry.id in selector.device_ids
+        ):
             selected.referenced_devices.add(device_entry.id)
-
     if not selector.area_ids and not selected.referenced_devices:
         return selected
 
     for ent_entry in ent_reg.entities.values():
-        if ent_entry.area_id in selector.area_ids or (
-            not ent_entry.area_id and ent_entry.device_id in selected.referenced_devices
+        if (
+            ent_entry.area_id in selector.area_ids
+            or (
+                not ent_entry.area_id
+                and ent_entry.device_id in selected.referenced_devices
+            )
+            or ent_entry.device_id in selector.device_ids
```

Comment:
This makes so much sense to me!

I wonder what the impact for the end-user would be. As in... 🤷